### PR TITLE
refactor(config)!: revamp the cursor configuration

### PR DESCRIPTION
**Pull request by orhun**
> [!WARNING]  
> this is hella breaking

Before:

```toml
cursor = '▇'
blinking-cursor = false
```

After:

```toml
[cursor]
shape = 'block'
blinking = false
```

### CODE REVIEW

**Comment by raphamorim** [comment]
I was going to write this code this week but you were quicker! 👀 

@orhun actually I think might be the best also change the style API to 

```toml
style = 'Block'
# or
style = 'Caret'
# or
style = 'underline'
```

What do you think?

**Comment by orhun** [comment]
yup, makes sense. Will get to it soon 👍🏼

**Comment by orhun** [comment]
I did you one better. Saw that there is already a `CursorShape` enum so I re-used it like so:

```toml
[cursor]
shape = 'block'
blinking = false
```

**Comment by orhun** [comment]
Can you test it before merge? For me the other shapes were not working as expected (even without this PR) - will look into it.

**Comment by raphamorim** [comment]
tested main, only `|` cursor working 👁️ 👁️ 

Screenshot 2024-09-05 at 19 39 19

**Comment by orhun** [comment]
yeah for me too. Something is freaky.

Will check it out

**Comment by raphamorim** [comment]
I figured out, I broke when I was refactoring SugarStyle into FragmentStyle lol
coming up a fix on main...

**Comment by raphamorim** [comment]
Should be fixed now 

Screenshot 2024-09-05 at 20 02 56

File: rio-backend/src/ansi/mod.rs [hunk 1]
```diff
@@ -27,6 +34,16 @@ impl CursorShape {
     }
 }
 
+impl From<CursorShape> for char {
+    fn from(value: CursorShape) -> Self {
+        match value {
+            CursorShape::Underline => '_',
+            CursorShape::Beam => '|',
+            _ => '▒',
```
Comment (raphamorim) [inline rio-backend/src/ansi/mod.rs:42]: I think the character is `'▇'`

Comment (raphamorim) [reply rio-backend/src/ansi/mod.rs:42]: actually we can get rid of this `From<CursorShape> for char`

Comment (raphamorim) [reply rio-backend/src/ansi/mod.rs:42]: and doesn't matter though which character, since it always fallbacks to block

Comment (orhun) [reply rio-backend/src/ansi/mod.rs]: 58b0c1b3094f1085e12a7827ab7e08495fd36aa3

**Review by raphamorim — APPROVED** [review]
🚀 ⭐

**Pull request merged**